### DISCUSSION TITLE
ext:static in route

### DISCUSSION THREAD

**Discussion by samueller** [AnswerDotAI/fasthtml#356] [General]
Docs show the following code for [changing static media file locations](https://docs.fastht.ml/tutorials/quickstart_for_web_devs.html#other-static-media-file-locations):
```python
@rt("/{fname:path}.{ext:static}")
async def get(fname:str, ext:str): 
    return FileResponse(f'public/{fname}.{ext}')
```

I added that route and put my static content in `public/`, but I only got `Not Found`s. When I remove `:static` from the route decorator, it worked. Did I do something wrong? Was `:static` a typo? What is `:path`?

**Comment by zholmquist** [comment]
Started using FastHTML today, and ran into this wall fast. Got it to work with the following code:

```python
from pathlib import Path

@rt("/{fname:path}.{ext:static}")
async def get(fname: str, ext: str):
    current_dir = Path(__file__).resolve().parent
    file_path = current_dir / "public" / f"{fname}.{ext}"
    return FileResponse(file_path)
 ```
